ignite/pkg/cosmosgen/templates/module/module.ts.tpl

```diff
@@ -0,0 +1,71 @@
+// Generated by Ignite ignite.com/cli
+
+import { StdFee } from "@cosmjs/launchpad";
+import { SigningStargateClient, DeliverTxResponse } from "@cosmjs/stargate";
+import { EncodeObject } from "@cosmjs/proto-signing";
+
+import { Api } from "./rest";
+{{ range .Module.Msgs }}import { {{ .Name }} } from "./types/{{ resolveFile .FilePath }}";
+{{ end }}
+
+{{ range .Module.Msgs }}
+type send{{ .Name }}Params = {
+  value: {{ .Name }},
+  fee?: StdFee,
+  memo?: string
+};
+{{ end }}
+{{ range .Module.Msgs }}
+type {{ camelCase .Name }}Params = {
+  value: {{ .Name }},
+};
+{{ end }}
+
+class Module extends Api<any> {
+	private _client: SigningStargateClient;
```

Comment:
If I want to sign and broadcast myself let's say in the bank module, I can call `Module.msgSend` to get the message but how to sign and broadcast? `Module._client` is a private prop.

Reply:
`Module` has `withSigner` to attach a signer.

Reply:
> `typescript` alone doesn't mean anything really it's just the name of underlying language,
> 
>     * we could do `ts-client` or `ts-sdk`,
> 
>     * and `vue-store` or something else

Good point. I'm okay w/ `ignite g ts-client` and `ignite g vue-utils`.

Reply:
How can I sign and broadcast multiple messages, can you please add a sample snippet?

Reply:
Just:

```ts
let msgs = [p1, p2, p3].map(item => ignite.cosmosBank.msgSend(item))
ignite.signer.client.signAndBroadcast(msg)
```
  
 As in [here](https://github.com/ignite-hq/ignite.com/blob/b2ab19ba2ee7c4de41ff8d73c4ee6eab86e4203f/src/components/requests/IgniteRequestsAcceptModal.vue#L156).